### ISSUE TITLE
Variable type duplicated when assigned using isinstance in ternary

### ISSUE THREAD

**Issue by audoh-tickitto** [microsoft/pylance-release#920]
## Environment data

-   Language Server version: 2021.2.0
-   OS and version: Windows 10 Pro 20H2 19042.789
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.1 acquired via pyenv

## Expected behaviour

- Type should only appear once

## Actual behaviour

- Type is duplicated:
![image](https://user-images.githubusercontent.com/71696787/107025443-5cd83180-67a1-11eb-85d7-83cab6450360.png)

## Logs

N/A 
(I enabled logging but didn't see any relevant output in the panel when entering the problem code; when I saved I just saw the commands run for the linter, type checker etc. I have set up)

## Code Snippet / Additional information

```python
import json
json_path = "my_json.json"
with open(json_path, "rb") as data_file:
    parsed = json.load(data_file, object_hook=json_util.object_hook)
    docs = parsed if isinstance(parsed, list) else [parsed]
```

**Comment by erictraut** [comment]
What's happening here is that the types in the union are `list[Any]` and `list[Unknown]`. But because you have type checking disabled, we hide any `Any` and `Unknown` type arguments under the assumption that you don't want to see those additional details. I'll look at how we can elide the seemingly-redundant type from the union when displaying it.

**Comment by audoh-tickitto** [comment]
Cool, I didn't even realise that was off by default but did wonder why I didn't see generic types! I've just enabled it.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.2.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202121-10-february-2021
